api/v1/user_types.go

```diff
@@ -0,0 +1,69 @@
+/*
+MIT License
+
+Copyright (c) 2021 StorageOS
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// UserSpec defines the desired state of User.
+type UserSpec struct {
+	// SecretRefName is the name of a Secret that contains the user
+	// credentials.
+	SecretRefName string `json:"secretRefName,omitempty"`
+	// IsAdmin is used to set a user as an admin.
+	IsAdmin bool `json:"isAdmin,omitempty"`
```

Comment:
Does it make sense to create a separated `Role` type where admin is only one role? This looks not so flexible.

Reply:
I'm not sure about that. I don't think there are any other roles other than admin and non-admin. There's group with policies to create restricted groups, but that's different.